src/mlpack/tests/CMakeLists.txt

```diff
@@ -126,8 +122,24 @@ add_custom_command(TARGET mlpack_test
   WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
 )
 
-# For 'make test'.
-add_test (NAME mlpack_test
-  COMMAND mlpack_test "--log_level=test_suite" # Set UTF runtime param
-  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/ # This is where test files are put.
-)
+# Add tests to the testing framework
+get_target_property(test_sources mlpack_test SOURCES)
+
+foreach(test_file ${test_sources})
```

Comment:
It would be great if you could add some comments here, might be helpful for someone trying to understand the construct.